python/kwiver/vital/types/metadata.cxx

```diff
@@ -18,61 +18,94 @@
 namespace py = pybind11;
 using namespace kwiver::vital;
 
-void adder(metadata &self, py::object data, vital_metadata_tag t )
+kwiver::vital::any
+py_object_to_any( vital_metadata_tag tag, py::object data )
 {
-
-  #define TAG_CASE( TAG, NAME, TYPE, ... ) case VITAL_META_##TAG: \
-  { \
-  auto vital_meta_data_ ## TAG = data.cast< typename vital_meta_trait< VITAL_META_ ## TAG >::type >(); \
-  self.add<VITAL_META_ ## TAG>(vital_meta_data_ ## TAG); \
-  break; \
+  auto const& type = tag_traits_by_tag( tag ).type();
+  if( type == typeid( bool ) )
+  {
+    return data.cast< bool >();
   }
-
-  switch (t)
+  else if ( type == typeid( uint64_t ) )
```

Comment:
Curious why the change from a switch statement to an if-else structure.

Reply:
Previously, it was switching on the metadata tag. The `KWIVER_VITAL_METADATA_TAGS` macro was used to generate the ~100 case statements for each of the tags, which was necessary because we don't know the internal type of `data`, but that can be statically determined from the tag. However, the major change of this pull request removed that macro. We can still achieve the same behavior by checking all the possible types of `data` manually, as we do here. It is a little ugly (as was the macro), but as I mentioned above I am planning a later change which would allow a more elegant way to go through all supported types, so ideally this is just temporary until then.